backend/models/user.model.test.js

```diff
@@ -0,0 +1,165 @@
+        desiredJobTitle: 'Senior Software Engineer',
+        skillsToMatch: ['Jest', 'Node.js'],
+        firstAttended: '2025-01-01',
+        attendanceReason: 'To learn and contribute',
+        githubHandle: 'mockuser',
+        projects: ['ProjectId1', 'ProjectId2'],
+        phone: '123-456-7890',
+        textingOk: true,
+        slackName: 'mockuser',
+        isHflaGithubMember: true,
+        githubPublic2FA: true,
+        availability: 'Weekdays',
+        managedProjects: ['Project1', 'Project2'],
+        isActive: true,
+      };
+
+      // Create a mock user instance
+      const mockUser = new User(userObj);
+      const serializedUser = mockUser.serialize();
+
+      // Test
+      expect(serializedUser).toEqual({
+        id: mockUser._id,
+        name: {
+          firstName: mockUser.name.firstName,
+          lastName: mockUser.name.lastName,
+        },
+        email: mockUser.email,
+        accessLevel: mockUser.accessLevel,
+        createdDate: mockUser.createdDate,
+        currentRole: mockUser.currentRole,
+        desiredRole: mockUser.desiredRole,
+        newMember: mockUser.newMember,
+        currentJobTitle: mockUser.currentRole,
+        desiredJobTitle: mockUser.desiredRole,
+        skillsToMatch: mockUser.skillsToMatch,
+        firstAttended: mockUser.firstAttended,
+        attendanceReason: mockUser.attendanceReason,
+        githubHandle: mockUser.githubHandle,
+        projects: mockUser.projects,
+        phone: mockUser.phone,
+        textingOk: mockUser.textingOk,
+        slackName: mockUser.slackName,
+        isHflaGithubMember: mockUser.isHflaGithubMember,
+        githubPublic2FA: mockUser.githubPublic2FA,
+        availability: mockUser.availability,
+        managedProjects: mockUser.managedProjects,
+        isActive: mockUser.isActive,
+      });
+    });
+  });
+
+  describe('Validation test', () => {
+    it('should fail validation check if user email is not unique', async () => {
+      // Create and save a mock user with a specific email
+      const mockUser1 = new User({ email: 'mockuser@example.com' });
+      await mockUser1.save();
+
+      // Create another mock user with the same email
+      const mockUser2 = new User({ email: 'mockuser@example.com' });
+
+      // Attempt to validate the mock user 2 by checking for uniqueness in email
+      let error;
+      try {
+        await mockUser2.save();
+      } catch (err) {
+        error = err;
+      }
+
+      // Tests
+      expect(error).toBeDefined();
+      expect(error.code).toBe(11000);
+    });
+
+    it('should fail validation check if accessLevel is invalid', async () => {
+      // Create a mock user with an invalid accessLevel
+      const mockuser = new User({
+        accessLevel: 'projectleader', // not 'user', 'admin', or 'superadmin'
+      });
+
+      // Attempt to validate the mock user by checking for valid accessLevel
+      let error;
+      try {
+        await mockuser.save();
```

Comment:
This should be changed to `await mockuser.validate();` so as to not require a mongo server